ctest-next/tests/basic.rs

```diff
@@ -0,0 +1,13 @@
+use std::error::Error;
+
+use ctest_next::TestGenerator;
+
+#[test]
+fn test_entrypoint() -> Result<(), Box<dyn Error>> {
+    let generator = TestGenerator::new();
+
+    generator.generate("./tests/sources/hierarchy/lib.rs", "hierarchy_out.rs")?;
+    generator.generate("./tests/sources/simple.rs", "simple_out.rs")?;
```

Comment:
`?` -> `.unwrap()` so we get the information about which line failed. The test can drop the `-> Result<...>` return type.

Could you also split these test sources into two different `#[test]` functions?